addons/pos_coupon/static/src/js/tours/PosCoupon4.tour.js

```diff
@@ -25,4 +26,22 @@ odoo.define('pos_coupon.tour.pos_coupon4', function (require) {
 
 
     Tour.register('PosCouponTour4', { test: true, url: '/pos/web' }, getSteps());
+
+    startSteps();
+
+    ProductScreen.do.clickHomeCategory();
+
+    ProductScreen.exec.addOrderline('Test Product 1', '1');
+    ProductScreen.exec.addOrderline('Test Product 2', '1');
+    ProductScreen.do.clickPricelistButton();
+    ProductScreen.do.selectPriceList('Public Pricelist');
+    PosCoupon.do.enterCode('abcdb');
```

Comment:
May I know what's the total before entering the code?
Also, what is due before setting the customer?

Reply:
Before applying the coupon code the total is 303,75€ (for some reason it's adding 11 product 2) and the due is 0 € before selecting the customer. Should I add those steps in the tour ?

Reply:
11 product 2? Hmm. A good practice is to make a check (say the price of the newly added line) before adding new line. In your case, you called `addOrderline` one after the other, so the `pressing` of digit `1` is made for product 2 resulting to quantity 11. A safer alternative to adding "1" quantity of a product is using `do.clickProduct`.

Yes, please add the expected amount before and after selecting the customer in the payment screen so that we know whether the fix works or not.